### DISCUSSION TITLE
Mutate data for table

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#6523] [Help] [answered]
Can I mutate data before it's shown in a table?  For example,

`     TextColumn::make('customer.created_at')->label("Customer Date"),     
`

will display the usual date formatting but what if I want to apply a  custom date format?

In other weird cases I have Model methods that return JSON but I'd like to use a closure to parse only certain data before it gets displayed.

And also, some of my data is a URL. Is there an easy way to make it a clickable anchor tag in the table? Or do I need a custom view to do that?

Thanks everyone!

**Comment by danharrin** [comment, marked as answer]
https://filamentphp.com/docs/2.x/tables/columns/text#custom-formatting

https://filamentphp.com/docs/2.x/tables/columns/getting-started#opening-urls
